### PR TITLE
PIM-11103: Fix deletion of enriched category attribute when deleting attribute

### PR DESCRIPTION
**Pull request by jeremyBeaucousinAkeneo**
…attribute

**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSql.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+use Akeneo\Category\Application\Query\GetDeactivatedAttribute;
+use Akeneo\Category\Domain\Model\Attribute\Attribute;
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeCollection;
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeUuid;
+use Akeneo\Category\Domain\ValueObject\Template\TemplateUuid;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Exception;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSql.php:16]:
```suggestion
 * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
```

File: src/Akeneo/Category/back/Application/Query/GetDeactivatedAttribute.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types=1);
+
+namespace Akeneo\Category\Application\Query;
+
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeCollection;
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeUuid;
+use Akeneo\Category\Domain\ValueObject\Template\TemplateUuid;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Application/Query/GetDeactivatedAttribute.php:12]:
```suggestion
 * @copyright 2023 Akeneo SAS (http://www.akeneo.com)
```

File: src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSqlIntegration.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+use Akeneo\Category\Application\Query\GetAttribute;
+use Akeneo\Category\Application\Query\GetDeactivatedAttribute;
+use Akeneo\Category\back\tests\Integration\Helper\CategoryTestCase;
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeCollection;
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeUuid;
+use Akeneo\Category\Domain\ValueObject\Template\TemplateUuid;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSqlIntegration.php:15]:
```suggestion
 * @copyright 2023 Akeneo SAS (http://www.akeneo.com)
```

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSql.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    public function __construct(private Connection $connection)
+    {
+    }
+
+    /**
+     * @throws Exception
+     * @throws \JsonException
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function byTemplateUuid(TemplateUuid $uuid): AttributeCollection
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSql.php:30]: Is this method used anywhere?

File: src/Akeneo/Category/back/tests/Integration/Application/Command/CleanCategoryTemplateAttributeAndEnrichedValues/CleanCategoryTemplateAttributeAndEnrichedValuesCommandHandlerIntegration.php [hunk 5]
```diff
@@ -87,4 +106,15 @@ private function updateCategoryValues(string $code, string $channel = 'ecommerce
...
+    protected function deactivateAttribute(string $uuid): void
+    {
+        $query = <<<SQL
+            UPDATE pim_catalog_category_attribute SET is_deactivated = 1 WHERE uuid = :uuid;
+        SQL;
+
+        $this->connection->executeQuery($query, [
+            'uuid' => Uuid::fromString($uuid)->getBytes(),
+        ]);
+    }
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/tests/Integration/Application/Command/CleanCategoryTemplateAttributeAndEnrichedValues/CleanCategoryTemplateAttributeAndEnrichedValuesCommandHandlerIntegration.php:119]: There is a helper method for that in CategoryTestCase

```suggestion
```

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSql.php (hunk 4 above)
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetDeactivatedCategoryTemplateAttributeSql.php:30]: No it isn't used, I implemented it thinking I'll need it, I forgot to remove it.

**Review by etiennekerbastard — APPROVED** [review]

**Review by pchasle — APPROVED** [review]
:rocket:

**Pull request merged**